pkg/clusteragent/orchestrator/controller.go

```diff
@@ -128,23 +144,70 @@ func (o *Controller) Run(stopCh <-chan struct{}) {
 		return
 	}
 
-	if !cache.WaitForCacheSync(stopCh, o.unassignedPodListerSync) {
+	if !cache.WaitForCacheSync(stopCh, o.unassignedPodListerSync, o.deployListerSync, o.rsListerSync) {
 		return
 	}
 
 	go wait.Until(o.processPods, 10*time.Second, stopCh)
+	// spread the processing of other resources
+	time.AfterFunc(2*time.Second, func() {
```

Comment:
my 2 cents: maybe you can create a channel `processPodsSync` (kind of `cache.InformerSynced`) to execute the `processReplicaSets` only the `processPods` have been executed once.

Reply:
that could be a potential improvement indeed, I'll keep it in my mind when we'll need a more advanced dispatching